client/client.go

```diff
@@ -43,6 +43,11 @@ import (
 	"sigs.k8s.io/boskos/storage"
 )
 
+const (
+	// resourceTypeErrorSnippet is present in the HTTP response if user requests a resource type that does not exist.
+	resourceTypeErrorSnippet = "resource type"
```

Comment:
this is a pretty loose contract imo. The error message is "fmt.Sprintf("resource type %q does not exist", r.rType)" according to https://github.com/kubernetes-sigs/boskos/blob/9fae38623263fd9251734b5dd5d78a0f09c3876a/ranch/ranch.go#L62. A couple of thoughts here:

- Option 1: since the rType is known by the time the condition is checked, I think we can do strict full string matching.
- Option 2: add a function directly in `rancher`, side by side with `func (r ResourceTypeNotFound) Error()` function, called something like `func isResourceTypeNotFoundErrorString`, and add a unit test to ensure that these two functions always agree with each other

Reply:
I've implemented something close to Option 2 where there is only "once source of truth" for the message shared by the client and server - defined in common.go